react-native/src/status_im/react_native/js_dependencies.cljs

```diff
@@ -1,31 +1,55 @@
 (ns status-im.react-native.js-dependencies)
 
-(def camera                 (js/require "react-native-camera"))
 (def config                 (js/require "react-native-config"))
-(def dialogs                (js/require "react-native-dialogs"))
-(def dismiss-keyboard       (js/require "dismissKeyboard"))
 (def fs                     (js/require "react-native-fs"))
 (def http-bridge            (js/require "react-native-http-bridge"))
 ;; i18n is now exported in default object of the module
 ;; https://github.com/AlexanderZaytsev/react-native-i18n/blob/master/index.js
-(def i18n                   (.-default (js/require "react-native-i18n")))
-(def image-crop-picker      (js/require "react-native-image-crop-picker"))
-(def image-resizer          (js/require "react-native-image-resizer"))
-(def instabug               (js/require "instabug-reactnative"))
 (def keychain               (js/require "react-native-keychain"))
-(def nfc                    (js/require "nfc-react-native"))
 (def qr-code                (js/require "react-native-qrcode"))
 (def react-native           (js/require "react-native"))
 (def realm                  (js/require "realm"))
 (def webview-bridge         (js/require "react-native-webview-bridge"))
-(def svg                    (js/require "react-native-svg"))
-(def react-native-fcm       (js/require "react-native-fcm"))
 (def secure-random          (.-generateSecureRandom (js/require "react-native-securerandom")))
-(def snoopy                 (js/require "rn-snoopy"))
-(def snoopy-filter          (js/require "rn-snoopy/stream/filter"))
-(def snoopy-bars            (js/require "rn-snoopy/stream/bars"))
-(def snoopy-buffer          (js/require "rn-snoopy/stream/buffer"))
 (def EventEmmiter           (js/require "react-native/Libraries/vendor/emitter/EventEmitter"))
-(def background-timer       (.-default (js/require "react-native-background-timer")))
 (def fetch                  (.-default (js/require "react-native-fetch-polyfill")))
-(def testfairy              (js/require "react-native-testfairy"))
+
+;; We don't use platform/desktop? here because that would lead to a circular dependency
+;; since platform.cljs itself depends on js-dependencies
+(def desktopPlatform?
+    (= (.-OS (.-Platform react-native)) "desktop"))
+
+(if desktopPlatform?
+  (do (def i18n                   (js/require "react-native-i18n"))
```

Comment:
It feels like it would be clearer to split up desktop and mobile `defs` here into separate files and bring them into this namespace via `: refer :all` rather than having `do` perform these definitions via side effects.

Reply:
@rcullito, thanks, I've created new PR for these changes, will fix your note there - #5203